### DISCUSSION TITLE
Add some PAPR reduction scheme

### DISCUSSION THREAD

**Discussion by jmso84** [jgaeddert/liquid-dsp#321] [Ideas]
I used liquid-dsp a few years ago to transfer low bitrate data between two custom HW and I remember high PAPR was killing me. It would be nice to add some PAPR mitigation technique for OFDM.

**Comment by jgaeddert** [comment]
Ah, yes. So I know of essentially 4 possibilities for addressing PAPR in OFDM, specifically:

1. Sacrifice a certain number of subcarriers to reduce PAPR by suppressing peaks in the time-domain. This is orthogonal in that it doesn't degrade the signal in the data-carrying subcarriers, but it is both computationally expensive to perform the search, and you lose some of your data rate by sacrificing some of your bandwidth.
2. Perform a pulse-like kernel to iteratively suppress peaks in the time-domain. This doesn't sacrifice any subcarriers, and doesn't impact the occupied spectrum, but it does distort the data-carrying subcarriers and can increase errors. It's also iterative and can take some computation.
3. Compress the transmitted signal. By far the easiest, but not spectrally clean, and distorts the signal (inter-channel interference). Might be possible to invert the compression on the receive side, but this hasn't been shown to be very effective.
4. If you have multiple phase-coherent transmit paths, you can split the subcarriers between them. For example, half of your subcarriers get transmitted out of DAC A while the other half go out DAC B. Obviously requires some more customized hardware, and can be computationally expensive to determine which subcarriers should be allocated to which transmit path.

There might also be some clever tricks you can play with adjusting the phase/amplitude of subcarriers with a P/N sequence, possibly conveyed over a side-channel, but this gets really customized and complicated really quickly.

My first thought would be to extend the [ofdmflexframe](https://liquidsdr.org/doc/ofdmflexframe/) to include a fourth subcarrier allocation type for PAPR reduction, addressing 1, above. Thoughts?
